lib/facility.rb

```diff
@@ -1,14 +1,75 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
-
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  attr_reader :name, 
+              :address, 
+              :phone, 
+              :services,
+              :registered_vehicles
+  
+  def initialize(facility_data)
+    @name = facility_data[:name] 
+    @address = facility_data[:address] 
+    @phone = facility_data[:phone] 
     @services = []
+    @registered_vehicles = []
   end
-
-  def add_services(service)
+  
+  def add_service(service)
     @services << service
   end
+
+  def collected_fees
+    @collected_fees = 0
+    @registered_vehicles.map do |vehicle|
+      if vehicle.plate_type == :antique
+        @collected_fees += 25
+      elsif vehicle.plate_type == :ev
+        @collected_fees += 200
+      else vehicle.plate_type == :regular
+        @collected_fees += 100
+      end
+    end
+    @collected_fees.to_i
+  end
+
+  def register_vehicle(car)
+    if @services.include?('Vehicle Registration')
+      @registered_vehicles << car
+    end
+    car.registration_date = Date.today
+      if car.antique? == true
+        car.plate_type = :antique
+      elsif car.engine == :ev
+        car.plate_type = :ev
+      else
+        car.plate_type = :regular
+      end
+  end
+
+  def administer_written_test(registrant)
+    if @services.include?('Written Test') 
+      if registrant.age >= 16 && registrant.permit? 
+        registrant.license_data[:written] = true
+      end
+    end 
+    registrant.license_data[:written]
+  end
```

Comment:
For this method and the similar ones below, you could refactor to 'flatten' the conditional logic:
```
if @services.include?('Written Test') && registrant.age >= 16 && registrant.permit?
  registrant.license_data[:written] = true
end
registrant.license_data[:written]
```
Of course, that line gets long. Maybe it warrants a helper method.
```
def written_eligible?(registrant)
  registrant.age >= 16 && registrant.permit?
end

def administer_written_test(registrant)
  if @services.include?('Written Test') && written_eligible?(registrant)
    ...
  end
  registrant.license_data[:written]
end
```